### DISCUSSION TITLE
OpenID using Defguard

### DISCUSSION THREAD

**Discussion by Zekhap** [9p4/jellyfin-plugin-sso#176] [General]
Hello, so ive been trying to use Defguard for the OpenID the past days.
But i always get the response "access_denied Try logging in again." in my webbrowser
And my console says 
"SSO Controller initialized" and "CustomAuthentication was not authenticated. Failure message: Invalid token."

But i dont know that it means by invalid token.

Did i do something wrong with my setup?

I can provide more information if you need.

**Comment by 9p4** [comment]
Can you upload both the Defguard configuration files as well as the plugin's XML configuration so I can try to reproduce the error?

**Reply by Zekhap** [reply]
`{"defguard":{"OidEndpoint":"https://defguard.example.com/","OidClientId":"123123123123","OidSecret":"123123123123123123","Enabled":true,"EnableAuthorization":true,"EnableAllFolders":false,"EnabledFolders":[],"AdminRoles":[],"Roles":[],"EnableFolderRoles":false,"EnableLiveTvRoles":false,"EnableLiveTv":false,"EnableLiveTvManagement":false,"LiveTvRoles":[],"LiveTvManagementRoles":[],"FolderRoleMapping":[],"OidScopes":[],"NewPath":true,"CanonicalLinks":{},"DisableHttps":false,"DoNotValidateEndpoints":false,"DoNotValidateIssuerName":false}}`

Random info.
1. Go to website
2. Click Login with SSO href="https://jellyfin.example.com/sso/OID/start/defguard"
3. Website says "access_denied Try logging in again." with url https://jellyfin.example.com/sso/OID/redirect/defguard?error=access_denied&state=somerandomcharacters here

For the installation i just installed the plugin
rebooted
went to the plugins and clicked on SSO-Auth and filled in the options.

Inside Defguard
i just created a new App and 
Redirect URL: https://jellyfin.example.com/sso/OID/redirect/defguard
with all scopes.

Is this what you meant?

**Reply by 9p4** [reply]
Jellyfin should have an XML file called `SSO-Auth.xml` in the configuration directory. Could you upload that?

**Reply by Zekhap** [reply]
This one?

`<?xml version="1.0" encoding="utf-8"?>
<PluginConfiguration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <SamlConfigs />
  <OidConfigs>
    <item>
      <key>
        <string>defguard</string>
      </key>
      <value>
        <PluginConfiguration>
          <OidEndpoint>https://defguard.example.com/</OidEndpoint>
          <OidClientId>12123123123123</OidClientId>
          <OidSecret>123123123123123123</OidSecret>
          <Enabled>true</Enabled>
          <EnableAuthorization>true</EnableAuthorization>
          <EnableAllFolders>false</EnableAllFolders>
          <EnabledFolders />
          <AdminRoles />
          <Roles />
          <EnableFolderRoles>false</EnableFolderRoles>
          <EnableLiveTvRoles>false</EnableLiveTvRoles>
          <EnableLiveTv>false</EnableLiveTv>
          <EnableLiveTvManagement>false</EnableLiveTvManagement>
          <LiveTvRoles />
          <LiveTvManagementRoles />
          <FolderRoleMappings />
          <OidScopes />
          <NewPath>true</NewPath>
          <CanonicalLinks />
          <DisableHttps>false</DisableHttps>
          <DoNotValidateEndpoints>false</DoNotValidateEndpoints>
          <DoNotValidateIssuerName>false</DoNotValidateIssuerName>
        </PluginConfiguration>
      </value>
    </item>
  </OidConfigs>`

**Reply by 9p4** [reply]
Yep. I'll set up Defguard and see if I can reproduce your issue.

**Reply by Zekhap** [reply]
okay, Was kinda annoying to setup Defguard. But without vpn setup it is probably easier.

**Reply by Zekhap** [reply]
How's it going 9p4?

**Reply by 9p4** [reply]
Sorry, it's exam season at the moment. When I find time I'll get back onto it :)
